Example/Example/CharacteristicsController.swift

```diff
@@ -49,16 +51,28 @@ class CharacteristicsController: UIViewController {
             }).addDisposableTo(disposeBag)
     }
 
-    private func showWriteFieldForCharacteristic(characteristic: Characteristic) {
+    fileprivate func showWriteFieldForCharacteristic(characteristic: Characteristic) {
         let valueWriteController = UIAlertController(title: "Write value", message: "Specify value in HEX to write ",
                                                      preferredStyle: .alert)
         valueWriteController.addTextField { textField in
-
+            textField.delegate = self
         }
         valueWriteController.addAction(UIAlertAction(title: "Cancel", style: .cancel, handler: nil))
         valueWriteController.addAction(UIAlertAction(title: "Write", style: .default) { _ in
-            print("")
+            
+            self.writeValueForCharacteristic(data: self.Data!, characteristic: characteristic)
```

Comment:
I believe here we could read the hex string: `valueWritrController.textFields.first?.text`. Remember to avoid force-unwrapping of optionals - let's stick to the safer versions - `if let` or `guard let`.